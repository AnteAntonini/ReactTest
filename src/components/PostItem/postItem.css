.post-section {
  display: flex;
  flex-direction: column;
  width: 80%;
  max-width: 800px;
  padding: 2rem 2.5rem;
  margin: 1rem 0;
  background-color: var(--color-martian-lightgray);
  border-radius: 4px;
}

.post-title {
  padding-bottom: 2rem;
  color: var(--color-martian-dark);
  font-size: 1.25rem;
  font-weight: 700;
}

.post-body {
  font-size: 0.875rem;
  color: var(--color-martian-darkgray);
  padding-bottom: 1.5rem;
}

.post-username {
  align-self: flex-end;
  color: var(--color-martian-darkgray);
  padding: 0.875rem 0;
}

.post-hover:hover {
  cursor: pointer;
  box-shadow: inset 3px 0 var(--color-martian-red);
}

.comment {
  padding: 1.5rem 0;
  font-size: 0.875rem;
  border-bottom: thin solid var(--color-martian-gray);
}

.comment:last-child {
  border: none;
}

.comment-title {
  padding: 1rem 0;
  color: var(--color-martian-dark);
  font-size: 1.25rem;
  font-weight: 600;
}

.comment-name {
  color: var(--color-martian-dark);
  font-weight: 600;
}

.comment-body {
  padding-top: 1.5rem;
  color: var(--color-martian-darkgray);
}
