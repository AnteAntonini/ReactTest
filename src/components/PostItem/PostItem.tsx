import { useEffect } from "react";
import "./postItem.css";
import { TPostsCommentsResponse } from "../../types/TComments";

type TPostItemParams = {
  id: number | string;
  postTitle: string;
  comments: TPostsCommentsResponse[] | [];
  message: string;
  postUserName?: string;
  postBody?: string;
  showCommentBody?: boolean;
  onClick?: any;
};

const PostItem = ({
  id,
  postTitle,
  postBody,
  postUserName,
  comments,
  showCommentBody = false,
  onClick,
  message,
}: TPostItemParams) => {
  useEffect(() => {
    console.log(`${message} Post Item`);
  }, []);

  return (
    <div
      key={id}
      className={`post-section ${onClick && "post-hover"}`}
      onClick={onClick}
    >
      <div className="post-title">{postTitle}</div>
      {postBody && <div className="post-body">{postBody}</div>}
      <span className="post-username">{postUserName}</span>
      <hr />
      <div>
        <h3 className="comment-title"> {comments?.length} Comments</h3>
        {comments.map((comment: TPostsCommentsResponse) => (
          <div key={comment.id} className="comment">
            <div className="comment-email">{comment.email}</div>
            <div className="comment-name">{comment.name}</div>
            {showCommentBody && (
              <div className="comment-body">{comment.body}</div>
            )}
          </div>
        ))}
      </div>
    </div>
  );
};

export default PostItem;
