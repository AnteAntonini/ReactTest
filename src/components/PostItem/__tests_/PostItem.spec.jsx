import { render } from "@testing-library/react";
import PostItem from "../PostItem";

describe("PostItem Component", () => {
  it("renders post item with all information", () => {
    const post = {
      id: 1,
      postTitle: "Sample Post",
      postBody: "This is the post body.",
      postUserName: "User123",
      comments: [
        {
          id: 1,
          email: "user@example.com",
          name: "User",
          body: "This is a comment",
        },
      ],
    };

    const { getByText } = render(<PostItem {...post} />);

    expect(getByText("Sample Post")).toBeInTheDocument();
    expect(getByText("This is the post body.")).toBeInTheDocument();
    expect(getByText("User123")).toBeInTheDocument();
    expect(getByText("1 Comments")).toBeInTheDocument();
    expect(getByText("user@example.com")).toBeInTheDocument();
    expect(getByText("User")).toBeInTheDocument();
  });

  it("renders post item with missing post body and username", () => {
    const post = {
      id: 2,
      postTitle: "Post Without Body and Username",
      comments: [
        {
          id: 2,
          email: "user2@example.com",
          name: "User2",
          body: "Another comment",
        },
      ],
    };

    const { getByText, queryByText } = render(<PostItem {...post} />);

    expect(getByText("Post Without Body and Username")).toBeInTheDocument();
    expect(queryByText("post-body")).toBeNull();
    expect(queryByText("post-username")).toBeNull();
    expect(getByText("1 Comments")).toBeInTheDocument();
    expect(getByText("user2@example.com")).toBeInTheDocument();
    expect(getByText("User2")).toBeInTheDocument();
  });

  it("renders post item with showCommentBody set to true", () => {
    const post = {
      id: 3,
      postTitle: "Post With Show Comment Body",
      showCommentBody: true,
      comments: [
        {
          id: 3,
          email: "user3@example.com",
          name: "User3",
          body: "Comment with body",
        },
      ],
    };

    const { getByText } = render(<PostItem {...post} />);

    expect(getByText("Post With Show Comment Body")).toBeInTheDocument();
    expect(getByText("1 Comments")).toBeInTheDocument();
    expect(getByText("user3@example.com")).toBeInTheDocument();
    expect(getByText("User3")).toBeInTheDocument();
  });
});

describe("Test snapshot", () => {
  it("Component should match snapshot", () => {
    const post = {
      id: 1,
      postTitle: "Sample Post",
      postBody: "This is the post body.",
      postUserName: "User123",
      comments: [
        {
          id: 1,
          email: "user@example.com",
          name: "User",
          body: "This is a comment",
        },
      ],
    };

    const { asFragment } = render(<PostItem {...post} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
