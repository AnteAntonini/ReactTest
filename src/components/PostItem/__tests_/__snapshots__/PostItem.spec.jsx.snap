// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-section undefined"
  >
    <div
      class="post-title"
    >
      Sample Post
    </div>
    <div
      class="post-body"
    >
      This is the post body.
    </div>
    <span
      class="post-username"
    >
      User123
    </span>
    <hr />
    <div>
      <h3
        class="comment-title"
      >
         1 Comments
      </h3>
      <div
        class="comment"
      >
        <div
          class="comment-email"
        >
          user@example.com
        </div>
        <div
          class="comment-name"
        >
          User
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
